azurelinuxagent/ga/exthandlers.py

```diff
@@ -350,6 +352,8 @@ def handle_ext_handler(self, ext_handler, etag):
             else:
                 message = u"Unknown ext handler state:{0}".format(state)
                 raise ExtensionError(message)
+        except RestartError:
+            raise
```

Comment:
Logging + telemetry needed here

Reply:
Added log statement here.

Would be better to do the logging and telemetry where the error actually gets processed?